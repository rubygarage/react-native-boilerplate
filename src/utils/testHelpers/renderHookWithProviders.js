/* eslint-disable import/no-extraneous-dependencies */
import { renderHook } from '@testing-library/react-hooks';
import { ThemeProvider } from 'styled-components';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import mockedTheme from 'utils/testHelpers/mockedTheme';

const store = configureStore()({});
store.dispatch = jest.fn();

const renderHookWithProviders = (hook) => {
  const wrapper = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={mockedTheme}>
        {children}
      </ThemeProvider>
    </Provider>
  );

  return renderHook(hook, { wrapper });
};

export default renderHookWithProviders;
