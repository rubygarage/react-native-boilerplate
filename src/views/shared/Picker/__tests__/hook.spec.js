import { act } from '@testing-library/react-hooks';

import renderHookWithProviders from 'utils/testHelpers/renderHookWithProviders';

import useContainer from '../hook';

describe('Picker useContainer hook', () => {
  const props = {
    selectedValue: 'test',
    options: [
      { value: 'test', label: 'label' },
      { value: 'test1', label: 'label1' },
    ],
    intl: {
      formatMessage: jest.fn(),
    },
    onValueChange: jest.fn(),
  };

  it('matches snapshot', () => {
    const { result } = renderHookWithProviders(() => useContainer(props));

    expect(result.current).toMatchSnapshot();
  });

  describe('checks detectLabel method', () => {
    it('should return right label', () => {
      const { result } = renderHookWithProviders(() => useContainer(props));

      let label = null;

      act(() => {
        label = result.current.detectLabel();
      });

      expect(label).toBe('label');
    });

    it('should return right label', () => {
      const { result } = renderHookWithProviders(() => useContainer({
        ...props,
        selectedValue: 'test1',
      }));

      let label = null;

      act(() => {
        label = result.current.detectLabel();
      });

      expect(label).toBe('label1');
    });

    it('should return undefined', () => {
      const { result } = renderHookWithProviders(() => useContainer({
        ...props,
        selectedValue: 'some',
      }));

      let label = null;

      act(() => {
        label = result.current.detectLabel();
      });

      expect(label).toBeFalsy();
    });
  });

  it('checks handlePickerToggle method', () => {
    const { result } = renderHookWithProviders(() => useContainer(props));

    act(() => {
      result.current.handlePickerToggle();
    });

    expect(result.current.isOpen).toBe(true);

    act(() => {
      result.current.handlePickerToggle();
    });

    expect(result.current.isOpen).toBe(false);
  });

  it('checks handleValueChange method', () => {
    const value = 'test';

    const { result } = renderHookWithProviders(() => useContainer(props));

    act(() => {
      result.current.handleValueChange(value);
    });

    expect(result.current.isOpen).toBe(true);
    expect(props.onValueChange).toHaveBeenCalledWith(value);
  });
});
