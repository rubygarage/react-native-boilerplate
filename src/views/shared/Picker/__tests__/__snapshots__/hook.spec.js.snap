// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker useContainer hook matches snapshot 1`] = `
Object {
  "detectLabel": [Function],
  "handlePickerToggle": [Function],
  "handleValueChange": [Function],
  "intl": Object {
    "formatMessage": [Function],
  },
  "isOpen": false,
  "setIsOpen": [Function],
  "theme": Object {
    "colors": null,
    "fonts": null,
    "sizes": null,
    "type": "Mocked theme",
  },
}
`;
